### PR TITLE
Upgrades operator-sdk version to v0.16.x

### PR DESCRIPTION
**Pull request by clcollins**
This commit upgrades the operator-sdk version to v0.16.x.
    
Includes changes introduced in:
    
* #499 (v0.10.x)
* #500 (v0.11.x)
* #501 (v0.12.x)
* #503 (v0.13.x)
* #504 (v0.14.x)
* #505 (v0.15.x)
    
This could be merged and the other PRs closed, or the changes can be merged one by one and the operator tested after each merge.

REF: [Upgrade aws-account-operator to sdk v0.16.x](https://issues.redhat.com/browse/OSD-5097)

    
Signed-off-by: Christopher Collins <collins.christopher@gmail.com>

### CODE REVIEW

File: pkg/apis/aws/v1alpha1/account_types.go [hunk 1]
```diff
@@ -50,8 +50,10 @@ type AccountStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
-	Claimed                  bool               `json:"claimed,omitempty"`
-	SupportCaseID            string             `json:"supportCaseID,omitempty"`
+	Claimed       bool   `json:"claimed,omitempty"`
+	SupportCaseID string `json:"supportCaseID,omitempty"`
+	// +listType=map
+	// +listMapKey=type`
```
Comment (2uasimojo) [inline pkg/apis/aws/v1alpha1/account_types.go:56]: I believe this can go away now.

File: pkg/apis/aws/v1alpha1/accountclaim_types.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ type AccountClaimSpec struct {
 // AccountClaimStatus defines the observed state of AccountClaim
 // +k8s:openapi-gen=true
 type AccountClaimStatus struct {
+	// +listType=map
+	// +listMapKey=type
```
Comment (2uasimojo) [inline pkg/apis/aws/v1alpha1/accountclaim_types.go:31]: ditto

File: pkg/apis/aws/v1alpha1/awsfederatedaccountaccess_types.go [hunk 3]
```diff
@@ -34,6 +34,8 @@ type AWSFederatedAccountAccessSpec struct {
 // AWSFederatedAccountAccessStatus defines the observed state of AWSFederatedAccountAccess
 // +k8s:openapi-gen=true
 type AWSFederatedAccountAccessStatus struct {
+	// +listType=map
+	// +listMapKey=type`
```
Comment (2uasimojo) [inline pkg/apis/aws/v1alpha1/awsfederatedaccountaccess_types.go:38]: ditto

File: pkg/apis/aws/v1alpha1/awsfederatedrole_types.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ type AWSFederatedRoleSpec struct {
 	AWSCustomPolicy AWSCustomPolicy `json:"awsCustomPolicy,omitempty"`
 	// AWSManagedPolicies is a list of amazong managed policies that exist in aws
 	// +optional
+	// +listType=atomic
```
Comment (2uasimojo) [inline pkg/apis/aws/v1alpha1/awsfederatedrole_types.go:33]: ditto

File: pkg/apis/aws/v1alpha1/awsfederatedrole_types.go [hunk 5]
```diff
@@ -64,7 +65,9 @@ type Condition struct {
 // AWSFederatedRoleStatus defines the observed state of AWSFederatedRole
 // +k8s:openapi-gen=true
 type AWSFederatedRoleStatus struct {
-	State      AWSFederatedRoleState       `json:"state"`
+	State AWSFederatedRoleState `json:"state"`
+	// +listType=map
+	// +listMapKey=type
```
Comment (2uasimojo) [inline pkg/apis/aws/v1alpha1/awsfederatedrole_types.go:70]: ditto

File: pkg/controller/accountpool/accountpool_controller_test.go [hunk 6]
```diff
@@ -199,10 +214,12 @@ func verifyAccountPool(c client.Client, expected *awsv1alpha1.AccountPool) bool
 	err := c.Get(context.TODO(), types.NamespacedName{Name: expected.Name, Namespace: expected.Namespace}, &ap)
 
 	if err != nil {
+		fmt.Printf("Error returning fakeclient accountPool: %s\n", err)
```
Comment (2uasimojo) [inline pkg/controller/accountpool/accountpool_controller_test.go:217]: Some of these changes look like linter fixes?

**Review by 2uasimojo — APPROVED** [review]
This looks legit to me, assuming it live-tests okay, including on 3.11 (to make sure we nailed that weird schema bug).

Minor cleanups noted inline, but nothing blocking that I can see.

/lgtm

**Review by iamkirkbater — APPROVED** [review]
/lgtm

Works on my machine :)

**Comment by rogbas** [comment]
/lgtm

Changes are looking good locally

**Pull request merged**